odoo/orm/fields_misc.py

```diff
@@ -121,7 +122,7 @@ def __set__(self, record, value):
         raise TypeError("field 'id' cannot be assigned")
 
     def convert_to_column(self, value, record, values=None, validate=True):
-        return int(value)
+        return value
```

Comment:
The existing code could spot programming mistakes.  It no longer will.

Reply:
Yes, it is used to generate the SQL for `IN` condition.
Now that we may have any type here, we won't spot silly mistakes like that.